motor/frameworks/asyncio.py

```diff
@@ -194,67 +191,35 @@ def settimeout(self, timeout):
     @asyncio_motor_sock_method
     @asyncio.coroutine
     def connect(self):
-        protocol_factory = lambda: self
-        # TODO: will call getaddrinfo again.
-        host, port = self.options.address
-        # socket module doesn't have an AF_UNIX constant on Windows.
         is_unix_socket = (self.options.family == getattr(socket,
                                                          'AF_UNIX', None))
         if is_unix_socket:
             path = self.options.address[0]
-            self._transport, protocol =(yield from self.loop.
-                create_unix_connection(protocol_factory, path,
-                                       ssl=self.ctx))
+            reader, writer = yield from asyncio.open_unix_connection(
+                path, loop=self.loop, ssl=self.ctx)
         else:
-            self._transport, protocol = yield from self.loop.create_connection(
-                protocol_factory, host, port,
-                ssl=self.ctx)
+            host, port = self.options.address
+            reader, writer = yield from asyncio.open_connection(
+                host=host, port=port, ssl=self.ctx, loop=self.loop)
+        self._reader, self._writer = reader, writer
 
     def sendall(self, data):
         assert greenlet.getcurrent().parent is not None,\
             "Should be on child greenlet"
 
         # TODO: backpressure? errors?
-        self._transport.write(data)
+        self._writer.write(data)
 
     @asyncio_motor_sock_method
     @asyncio.coroutine
     def recv(self, num_bytes):
-        while self._buffer_len < num_bytes:
-            yield from self._recv_future
-
-        data = bytes().join(self._buffer)
-        rv = data[:num_bytes]
-        remainder = data[num_bytes:]
-
-        self._buffer.clear()
-        if remainder:
-            self._buffer.append(remainder)
-
-        self._buffer_len = len(remainder)
-
+        rv = yield from self._reader.readexactly(num_bytes)
         return rv
 
     def close(self):
-        if self._transport:
-            self._transport.close()
-
-    # Protocol interface.
-    def connection_made(self, transport):
-        pass
-        # self._connected_future.set_result(None)
-
-    def data_received(self, data):
-        self._buffer_len += len(data)
-        self._buffer.append(data)
-
-        # TODO: comment
-        future = self._recv_future
-        self._recv_future = asyncio.Future(loop=self.loop)
-        future.set_result(None)
+        if self._writer:
+            self._writer.close()
 
-    def connection_lost(self, exc):
-        pass
```

Comment:
I guess that supposed to be implemented later, signaling via some sort of Future object